File: PersistenceAndServices/src/main/java/service_base/validators/ShowValidator.java
package service_base.validators;

import domain.Show;

public class ShowValidator implements Validator<Show> {
    @Override
    public void validate(Show entity) throws Exception {
        String errors="";

        if (entity.getAvailableSeats()==null)
            errors+="Seats are null!\n";
        if (entity.getAddress()==null)
            errors+="Address is null!\n";
        if (entity.getShowName()==null)
            errors+="Show name is null!\n";
        if (entity.getArtistName()==null)
            errors+="Artist name is null!\n";
        if (entity.getSoldSeats()==null)
            errors+="Sold seats is null!\n";
        if (entity.getStartTime()==null)
            errors+="Date is null!\n";
        if (entity.getAvailableSeats()<entity.getSoldSeats())
            errors+="Not enough seats left!\n";
        if (!errors.equals(""))
            throw new Exception("Invalid show!\n"+errors);
    }
}

File: MPPCS/Service/Validator/ShowValidator.cs
using CommonDomain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Persistence.Service.Validator
{
    internal class ShowValidator : Validator<Show>
    {
        public void validate(Show entity)
        {
            string errors = "";

            if (entity.GetAvailableSeats() < 0)
                errors += "Seats are invalid!\n";
            if (entity.GetArtistName() == "")
                errors += "Artist name is null!\n";
            if (entity.GetShowName() == "")
                errors += "Show name is null!\n";
            if (entity.GetSoldSeats() < 0)
                errors += "Sold seats are invalid!\n";
            if (entity.GetAddress() == "")
                errors += "Address is invalid!\n";
            if (entity.GetAvailableSeats() < entity.GetSoldSeats())
                errors += "Not enough seats left!\n";

            if (!errors.Equals(""))
                throw new Exception("Invalid show!\n" + errors);
        }
    }
}
